controller/controller.go

```diff
@@ -1019,18 +1010,10 @@ func ScaleScheduler(
 		)
 	} else {
 		logger.Infof("manually scaling scheduler %s to  %d GRUs", schedulerName, replicas)
-		var pods *v1.PodList
-		err := mr.WithSegment(models.SegmentPod, func() error {
-			var err error
-			pods, err = clientset.CoreV1().Pods(schedulerName).List(metav1.ListOptions{
-				LabelSelector: labels.Set{}.AsSelector().String(),
-				FieldSelector: fields.Everything().String(),
-			})
-			return err
-		})
+		// get list of actual pods
+		pods, err := ListCurrentPods(mr, clientset, schedulerName)
```

Comment:
Why do you still need to list pods here?

Reply:
I do. This is in `ScaleScheduler`. It is triggered by a manual scale request from the API

Reply:
It is inside the function that scales manually through API.

The API allows you to scale up/down and to a specific number of replicas. This pod listing is in the third case, as we need to know how many pods are running so we calculate the amount of pods to scale up or down to get the desired number

Reply:
I reckon this feature is not much used

Reply:
Okay. It makes sense to keep it like this.